src/test/java/utility/RegistrationFacade.java

```diff
@@ -0,0 +1,35 @@
+package utility;
+
+import com.github.javafaker.Faker;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import pages.SignUpPage;
+
+import java.time.Duration;
+
+public class RegistrationFacade {
```

Comment:
The facade should be a composition of other classes. In your case, it should be a composition of Faker, RegDataBuilder and SignUpPage classes.